src/App.js

```diff
@@ -1,24 +1,62 @@
-import logo from './logo.svg';
-import './App.css';
-
+import React, { useEffect, useState } from "react";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { addrecipe, addtypes } from "./redux/action/action";
+import Home from "./pages/home";
+import axios from "axios";
+import Recipedetail from "./components/recipedetail";
+import RecipeList from "./components/recipielist";
+import SearchPage from "./components/search.";
 function App() {
+  const dispatch = useDispatch();
+  const recipes = useSelector((state) => state.recipe);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  useEffect(() => {
+    const fetchRecipes = async () => {
+      try {
+        const reciperesponse = await axios.get("https://dummyjson.com/recipes");
+        const recipedata = reciperesponse.data;
+        dispatch(addrecipe(recipedata.recipes));
+        const catagoryresponse = await axios.get(
+          "https://dummyjson.com/recipes/meal-type/dinner"
+        );
+        const catagorydata = catagoryresponse.data;
+        dispatch(addtypes(catagorydata.recipes));
+      } catch (error) {
+        console.log("Error in fetching recipes: ", error);
```

Comment:
axios call should be inside of the action

https://www.digitalocean.com/community/tutorials/redux-redux-thunk

```
import {
  ADD_TODO_SUCCESS,
  ADD_TODO_FAILURE,
  ADD_TODO_STARTED,
  DELETE_TODO
} from './types';

import axios from 'axios';

export const addTodo = ({ title, userId }) => {
  return dispatch => {
    dispatch(addTodoStarted());

    axios
      .post(`https://jsonplaceholder.typicode.com/todos`, {
        title,
        userId,
        completed: false
      })
      .then(res => {
        dispatch(addTodoSuccess(res.data));
      })
      .catch(err => {
        dispatch(addTodoFailure(err.message));
      });
  };
};

const addTodoSuccess = todo => ({
  type: ADD_TODO_SUCCESS,
  payload: {
    ...todo
  }
});

const addTodoStarted = () => ({
  type: ADD_TODO_STARTED
});

const addTodoFailure = error => ({
  type: ADD_TODO_FAILURE,
  payload: {
    error
  }
});```